google-cloud-storage/src/test/java/com/google/cloud/storage/GrpcTransformPageDecoratorTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.paging.AbstractPage;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.PageContext;
+import com.google.api.gax.rpc.PagedListDescriptor;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.cloud.storage.GrpcStorageImpl.TransformingPageDecorator;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class GrpcTransformPageDecoratorTest {
+
+  @Test
+  public void valueTranslationTest() {
```

Comment:
Will you be adding a property based test in a follow-up PR?

Reply:
I can do that, I wanted to make a general test because I felt a property test would kind of just be duplicating the conversion tests, the only difference being wrapping the BlobInfo, BucketInfo etc in a Page but the translation logic is the same. Do you think there would be additional value to doing this?

Reply:
I was thinking you could use property based testing to generate the list of object names of different sizes (for initial Values, pageOne, pageTwo) then make then the test could do the rest that you're doing in this test. That was my only point in this comment.

Reply:
This test is specifically for the PageDecorator.

As far as testing the list calls themselves is concerned, I think those can be added in a later PR once we are also testing with the testbench.

Given that pages all have the same "shape" the things we care about testing there are:
1. Do we construct the request accurately
2. Do we wire in the correct decoder to the constructed TransformingPageDecorator

The decoder wiring part is pretty much verified for us at compile type via type checking.

The request construction part is probably more of a unit test, given how unfriendly the `*Options` stuff is.